src/mongoose_scram.erl

```diff
@@ -81,6 +82,13 @@ iterations() -> ?SCRAM_DEFAULT_ITERATION_COUNT.
 iterations(HostType) ->
     mongoose_config:get_opt([{auth, HostType}, password, scram_iterations]).
 
+password_to_scram(Password) ->
+    IterationCount = ?SCRAM_DEFAULT_ITERATION_COUNT,
+    ServerStoredKeys = [do_password_to_scram(Password, IterationCount, HashType)
+                        || {HashType, _Prefix} <- supported_sha_types()],
+    ResultList = lists:merge([{iteration_count, IterationCount}], ServerStoredKeys),
+    maps:from_list(ResultList).
+
```

Comment:
I think we don't need to store all the different SHA types here if ultimately we are just checking the password with the first one. Maybe make this function more configurable, pass the iteration sha types as a list? Also, maybe the function below (lines 95-101) could be converted into more generic helper, which would also accept the sha types as a list and allow for less copy-pasting?

Reply:
Ok, I rework this function to use only the given SHA type, and I use `sha512` for the domain admin password.